src/DocumentDB.ChangeFeedProcessor/Exceptions/PartitionException.cs

```diff
@@ -5,12 +5,20 @@
 namespace Microsoft.Azure.Documents.ChangeFeedProcessor.Exceptions
 {
     using System;
+    using System.Runtime.Serialization;
 
     /// <summary>
     /// General exception occurred during partition processing.
     /// </summary>
     public class PartitionException : Exception
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PartitionException" /> class.
+        /// </summary>
+        public PartitionException()
```

Comment:
Ctors in this class should probably be internal. Are consumers supposed to create instances of this class?

Reply:
Why don't we want consumers to be able to throw this type of exception?

Reply:
Would that be completely useless for us because we never catch this exception and will essentially treat it as "Exception". Same applies to PartitionNotFoundException, though for this one there one reason the consumers need to be able to create it: they would copy our implementation of partition processor which already throws that (which is kind of strange that it does BTW). Public API should be concise, if something is not used, it shouldn't be there, we can add it later when we start using it.

BTW, we need to have clear contract what kind of exceptions we expect from code implementing extensibility points. Like "if IPartitionProcessor is expected to throw PartitionSplitException when it detects split".